Commit new RFC for mono-repo gardening

amd-chrissosa opened the pull request:
This PR adds an RFC for gardening. The central doc is part of theRock with the intention of having each mono-repo have the actual rotation defined per repo. As the first rotation will be in rocm-libraries, my next PR will add the playbook there.

ScottTodd commented on docs/rfcs/RFC0002-MonoRepo-Gardener-Rotations.md:
```diff
@@ -0,0 +1,137 @@
+# Mono-repo Gardener
+
+Gardeners<sup>[1]</sup> are responsible for keeping their mono-repos shippable;
```
Do we want to qualify gardening with "mono-repo", especially when there are multiple "mono" repos? Maybe we could pick a different label like "component group build gardeners".

ScottTodd commented on docs/rfcs/RFC0002-MonoRepo-Gardener-Rotations.md:
```diff
@@ -0,0 +1,137 @@
+# Mono-repo Gardener
+
+Gardeners<sup>[1]</sup> are responsible for keeping their mono-repos shippable;
+ensuring bad commits get reverted by keeping build, test, and nightly release
+workflows green so that developers can remain productive and bugs are caught
+closer to commit times.
```
Phrasing as "bad commits" may be overly negative, especially if we don't define our terms carefully. I think this also inverts cause and effect: tree health is a factor of having automated workflows continually passing ("green"), and reverting "bad commits" quickly is one tool to achieve that. 

How about rewording to something along these lines instead?

> Gardeners are responsible for keeping their component areas continually shippable. Build, test, and nightly release workflows should be kept passing ("green"). When tests fail ("red") or builds fail ("black"), corrective action like identifying and reverting culprit commits should be taken quickly, so that developers can remain productive and bugs are caught closer to commit times.

ScottTodd commented on docs/rfcs/RFC0002-MonoRepo-Gardener-Rotations.md:
```diff
@@ -0,0 +1,137 @@
+In order to facilitate this work, gardeners have some super powers afforded to them
+that other developers do not such as ability to bypass any checks in CI.
+
+Note this concept isn't novel and can be referred to as build cops or build sheriffs
+in other projects. Good judgement and commitment are critical to being a good gardener.
```
nit: could wrap the other terms in quotes to highlight them (or italicize) 
```suggestion
Note this concept isn't novel and can be referred to as "build cops" or "build sheriffs"
in other projects. Good judgement and commitment are critical to being a good gardener.
```

(ah, and this was forked from a playbook that I wrote, funny reviewing docs I partly wrote myself 😅)

ScottTodd commented on docs/rfcs/RFC0002-MonoRepo-Gardener-Rotations.md:
```diff
@@ -0,0 +1,137 @@
+
+### Responsibilities and priorities
+
+1. When regressions occur that break CI workflows, revert those changes.
```
This could use preceding steps to

1. Communicate that something is broken
2. Identify the culprit commit(s)
3. Notify the authors/reviewers of those commits

ScottTodd commented on docs/rfcs/RFC0002-MonoRepo-Gardener-Rotations.md:
```diff
@@ -0,0 +1,137 @@
+   resolve, consider announcing that to affected developers and filing a
+   tracking issue on GitHub.
+
+Primary communication about the changes in a commit should happen on the pull
+request that commit came from. This ensures that members of the open source
+community (who may not be members of Discord or Slack servers) can see the
+discussion and avoid duplicating effort.
```
Could replace "Discord or Slack servers" with "other communication channels" (to cover Teams, email threads, etc. too)

ScottTodd commented on docs/rfcs/RFC0002-MonoRepo-Gardener-Rotations.md:
```diff
@@ -0,0 +1,137 @@
+* Reverts are generally safe, fast to make, and easy to review.
+* Fix-forwards can be tempting but carry additional risks and are necessarily
+  urgent to review.
+* When the shared `develop` branch is affected by a failing workflow, every
+  developer working off of that branch is affected. Reverting a change puts the
+  burden for fixing the issue back on just the original author of the change.
```
May want to emphasize the timeline and SLO here or elsewhere. I'd like to set expectations for turnaround time on the order of say... 30 minutes - 2 hours? Maybe with an escalation path for longer durations.

Could use a `[!IMPORTANT]` alert or some other styling: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts

ScottTodd reviewed (commented):
Nice, thank you! Generally LGTM, with a few suggestions for improvement.

stellaraccident commented on docs/rfcs/RFC0002-MonoRepo-Gardener-Rotations.md:
```diff
@@ -0,0 +1,137 @@
+workflows green so that developers can remain productive and bugs are caught
+closer to commit times.
+
+In order to facilitate this work, gardeners have some super powers afforded to them
```
Such as:

* The ability to bypass any checks in the CI.
* Authority to request that any developer on the project who is implicated in a breakage/degradation of service/resolution stop what they are doing and participate.
* In the event of major breakages, the ability to escalate and pull higher level management in to resolve issues.

ScottTodd commented on docs/rfcs/RFC0002-MonoRepo-Gardener-Rotations.md:
```diff
@@ -0,0 +1,137 @@
+# Mono-repo Gardener
+
+Gardeners<sup>[1]</sup> are responsible for keeping their mono-repos shippable;
```
Not quite what I had in mind for the footnote, since the "Google SWE Book:" item now has nothing under it:
![image](https://github.com/user-attachments/assets/f9388c34-5a6a-4b80-8015-804452fc352b)

ScottTodd reviewed (approved):
LGTM on my end. Probably want to address Stella's feedback and share with others who will join the rotation too.

amd-chrissosa commented:
> Please also apply the pre-commit formatting fixes after addressing comments / before merge: https://github.com/ROCm/TheRock/actions/runs/15888693201/job/44806750640?pr=917

Thanks totally missed the presubmit check! Mdformat ftw.
